packages/canon/src/theme/context.tsx

```diff
@@ -17,28 +17,59 @@
 import React, { createContext, useContext, ReactNode } from 'react';
 import { IconMap, IconNames } from '../components/Icon/types';
 import { defaultIcons } from '../components/Icon/icons';
+import { useMediaQuery } from '../hooks/useMediaQuery';
+import { Breakpoints } from '../types';
+
+const defaultBreakpoints: Breakpoints = {
+  xs: '0px',
+  sm: '640px',
+  md: '768px',
+  lg: '1024px',
+  xl: '1280px',
+  '2xl': '1536px',
+};
 
 interface ThemeContextProps {
   icons: IconMap;
+  breakpoint: keyof Breakpoints;
 }
 
 const ThemeContext = createContext<ThemeContextProps>({
   icons: defaultIcons,
+  breakpoint: 'md',
 });
 
+interface ThemeProviderProps {
+  children?: ReactNode;
+  overrides?: Partial<Record<IconNames, React.ComponentType>>;
+  breakpoints?: Partial<Breakpoints>;
+}
+
 /** @public */
-export const ThemeProvider = ({
-  children,
-  overrides,
-}: {
-  children: ReactNode;
-  overrides: Partial<Record<IconNames, React.ComponentType>>;
-}) => {
+export const ThemeProvider = (props: ThemeProviderProps) => {
+  const { children, overrides, breakpoints = defaultBreakpoints } = props;
+
   // Merge provided overrides with default icons
   const combinedIcons = { ...defaultIcons, ...overrides };
 
+  const isBreakpointSm = useMediaQuery(`(min-width: ${breakpoints.sm})`);
+  const isBreakpointMd = useMediaQuery(`(min-width: ${breakpoints.md})`);
+  const isBreakpointLg = useMediaQuery(`(min-width: ${breakpoints.lg})`);
+  const isBreakpointXl = useMediaQuery(`(min-width: ${breakpoints.xl})`);
+  const isBreakpoint2xl = useMediaQuery(`(min-width: ${breakpoints['2xl']})`);
+
+  // Determine the current breakpoint
+  const breakpoint = (() => {
+    if (isBreakpoint2xl) return '2xl';
+    if (isBreakpointXl) return 'xl';
+    if (isBreakpointLg) return 'lg';
+    if (isBreakpointMd) return 'md';
+    if (isBreakpointSm) return 'sm';
+    return 'xs';
+  })();
+
   return (
-    <ThemeContext.Provider value={{ icons: combinedIcons }}>
+    <ThemeContext.Provider value={{ icons: combinedIcons, breakpoint }}>
```

Comment:
hmm, not sure we want the theme provider to contain this kind of state, are there any other options here?

Reply:
Breakpoints is quite a big part of the process across the system to allow for responsive values like `variant={{ xs: 'small', md:'medium' }}`. The idea is to have a set list of keys but to allow for users to update them if they want to.

Why do you think we should not have it in the provider?

Reply:
I'm worried about the complexity of having this provider be stateful with a lot of built-in logic. Can leave as is for now an keep an eye on it